clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -179,8 +178,12 @@ public void ensureCoordinatorKnown() {
             RequestFuture<Void> future = sendGroupMetadataRequest();
             client.poll(future, requestTimeoutMs);
 
-            if (future.failed())
-                client.awaitMetadataUpdate();
+            if (future.failed()) {
+                if (future.isRetriable())
+                    client.awaitMetadataUpdate();
+                else
+                    throw future.exception();
```

Comment:
Yes, and I think that's what we want. For example, we need this check to propagate Authorization errors when the client is not authorized to the group. I will look over the code to see whether there are cases where we have a non-retriable exception which we wouldn't want to propagate here.